code/modules/hallucinations/effects/major.dm

```diff
@@ -408,3 +408,80 @@
 	if(images && images[1])
 		images[1].icon = 'icons/mob/alien.dmi'
 		images[1].icon_state = "alienh_pounce"
+
+/**
+  * # Hallucination - Sniper
+  *
+  * Fires a penetrator round at the target. On hit, knockdown + stam loss + hallucinated blood splatter for a bit.
+  */
+/obj/effect/hallucination/sniper
+	duration = 15 SECONDS
+
+/obj/effect/hallucination/sniper/Initialize(mapload, mob/living/carbon/target)
+	. = ..()
+
+	// Find a start spot for the sniper bullet
+	var/list/possible_turfs = list()
+	for(var/turf/t in RANGE_TURFS(13, target.loc))
+		possible_turfs += t
+	if(!length(possible_turfs))
+		return
+	var/turf/shot_loc = get_turf(pick(possible_turfs))
+	INVOKE_ASYNC(src, TYPE_PROC_REF(/obj/effect/hallucination/sniper, fire_bullet), shot_loc, target)
+
+/obj/effect/hallucination/sniper/proc/fire_bullet(turf/shot_loc, mob/living/carbon/target)
+	// Fire the bullet
+	var/obj/item/projectile/bullet/sniper/penetrator/hallucination/bullet = new(shot_loc)
+	bullet.hallucinator = target
+	bullet.def_zone = BODY_ZONE_HEAD
+	bullet.suppressed = TRUE
+
+	// Turn right away
+	var/matrix/M = new
+	var/angle = round(get_angle(shot_loc, target))
+	M.Turn(angle)
+	bullet.transform = M
+
+	// Handle who can see the bullet
+	if(target.client)
+		bullet.bullet_image = image(bullet.icon, bullet, bullet.icon_state, OBJ_LAYER, bullet.dir)
+		bullet.bullet_image.transform = M
+		target.client.images.Add(bullet.bullet_image)
+
+	// Start flying
+	bullet.trajectory = new(bullet.x, bullet.y, bullet.z, bullet.pixel_x, bullet.pixel_y, angle, SSprojectiles.global_pixel_speed)
+	bullet.last_projectile_move = world.time
+	bullet.has_been_fired = TRUE
+	target.playsound_local(target.loc, 'sound/weapons/gunshots/gunshot_sniper.ogg', 50)
```

Comment:
I think you have to make this play a sound directly into the target mind or client, otherwise other people can hear it

Reply:
playsound_local is used for many instances of target-only sounds, like modsuit activations and other hallucinations. My testing also has not shown the sound to be heard when out of body.

Reply:
`playsound_local` is indeed the correct proc here

Reply:
I am blind and didn't see the `_local` part